src/broker/Makefile.am

```diff
@@ -66,7 +66,8 @@ flux_broker_LDADD = \
 	$(builddir)/libbroker.la \
 	$(top_builddir)/src/common/libflux-core.la \
 	$(top_builddir)/src/common/libpmi/libpmi_client.la \
-	$(top_builddir)/src/common/libflux-internal.la
+	$(top_builddir)/src/common/libflux-internal.la \
+	$(top_builddir)/src/common/libflux-hostlist.la
```

Comment:
You might want to cherry-pick 454420e and 046ca14 and from #3276 instead, then whoever goes in second can rebase and drop.